edx_rest_framework_extensions/middleware.py

```diff
@@ -158,3 +159,83 @@ def _set_request_auth_type_metric(self, request):
         else:
             auth_type = 'session-or-unknown'
         monitoring.set_custom_metric('request_auth_type', auth_type)
+
+
+class JwtAuthCookieMiddleware(object):
+    """
+    Reconstitutes JWT auth cookies for use by API views which use the JwtAuthentication
+    authentication class.
+
+    We split the JWT across two separate cookies in the browser for security reasons. This
+    middleware reconstitutes the full JWT into a new cookie on the request object for use
+    by the JwtAuthentication class.
+
+    Also, sets the metric 'request_jwt_cookie' with one of the following values:
+        'yes': Value when reconstitution is successful.
+        'no': Value when both cookies are missing and reconstitution is not possible.
+        'missing-XXX': Value when one of the 2 required cookies is missing.  XXX will be
+            replaced by the cookie name, which may be set as a setting.  Defaults would
+            be 'missing-edx-jwt-cookie-header-payload' or 'missing-edx-jwt-cookie-signature'.
+
+    """
+    _JWT_DELIMITER = '.'
+
+    # The following are JWT_AUTH setting keys.
+    _JWT_AUTH_COOKIE_KEY = 'JWT_AUTH_COOKIE'
+    _JWT_AUTH_COOKIE_HEADER_PAYLOAD_KEY = 'JWT_AUTH_COOKIE_HEADER_PAYLOAD'
+    _JWT_AUTH_COOKIE_SIGNATURE_KEY = 'JWT_AUTH_COOKIE_SIGNATURE'
+
+    _JWT_COOKIE_NAME_DEFAULTS = {
+        _JWT_AUTH_COOKIE_KEY: 'edx-jwt-cookie',
+        _JWT_AUTH_COOKIE_HEADER_PAYLOAD_KEY: 'edx-jwt-cookie-header-payload',
+        _JWT_AUTH_COOKIE_SIGNATURE_KEY: 'edx-jwt-cookie-signature',
+    }
+
+    def _get_jwt_auth_cookie_setting(self, key):
+        """
+        Return the given JWT_AUTH setting value if set.  Otherwise, returns default.
+        """
+        value = settings.JWT_AUTH.get(key)
+        if value:
+            return value
+
+        return self._JWT_COOKIE_NAME_DEFAULTS[key]
```

Comment:
Should this be a `get` instead?  Since this is a generic helper, it's possible that our list of constants can become out of synch across accesses.